### PR TITLE
Added GPR publishing

### PR DESCRIPTION
**Pull request by georgedias**
Change workflow name to include action (from push-to-npm.yml to push-to-npm-gpr.yml)
Add the publishing to GitHub Package Registry (GPR)

### CODE REVIEW

File: .github/workflows/push-to-npm-gpr.yml [hunk 1]
```diff
@@ -28,3 +28,15 @@ jobs:
         run: npm publish --access public mitre-saf-v*.tgz
         env:
           NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+
+      # Setup .npmrc file to publish to GitHub Package Registry
```
Comment (Amndeep7) [inline .github/workflows/push-to-npm-gpr.yml:32]: where is the npmrc file?  what does it do?

File: .github/workflows/push-to-npm-gpr.yml [hunk 2]
```diff
@@ -28,3 +28,15 @@ jobs:
...
+      - uses: actions/setup-node@v3
+        with:
+          registry-url: 'https://npm.pkg.github.com'
```
Comment (Amndeep7) [inline .github/workflows/push-to-npm-gpr.yml:35]: if we're not using it to build but only publish, do we need to set the node-version attribute?  should we do it just to be on the safe side?

File: .github/workflows/push-to-npm-gpr.yml [hunk 3]
```diff
@@ -28,3 +28,15 @@ jobs:
...
+          scope: '@mitre'
```
Comment (Amndeep7) [inline .github/workflows/push-to-npm-gpr.yml:36]: can you gimme a documentation link for what scope does?  couldn't quickly find it on the github repo for this action.

in any case, why do we have it here and not on the one for the npm package as well?

File: .github/workflows/push-to-npm-gpr.yml [hunk 4]
```diff
@@ -28,3 +28,15 @@ jobs:
...
+
+      # Publish SAF CLI to GitHub Package Registry
+      - name: Publish SAF CLI to GPR
+        run: npm publish mitre-saf-v*.tgz
```
Comment (Amndeep7) [inline .github/workflows/push-to-npm-gpr.yml:40]: "When publishing scoped packages, the access level defaults to restricted. If you want your scoped package to be publicly viewable (and installable) set --access=public." - https://docs.npmjs.com/cli/v8/commands/npm-publish#access

idk why we need `--access public` for the npm one since it doesn't have that scope input, but shouldn't this publish call have it then?

File: .github/workflows/push-to-npm-gpr.yml
Comment (Amndeep7) [inline .github/workflows/push-to-npm-gpr.yml:1]: can we

- [ ] update the runs-on to a more up to date version of ubuntu
- [ ] make actions/checkout use v3 (which I think is the latest major version) instead of master
- [ ] make actions/setup-node for the npm release use v3 (which I think is the latest major version) instead of v1

**Review by Amndeep7 — CHANGES_REQUESTED** [review]

File: .github/workflows/push-to-npm-gpr.yml
Comment (georgedias) [reply .github/workflows/push-to-npm-gpr.yml:1]: done

**Review by georgedias — COMMENTED** [review]
Implemented recommendations

**Review by Amndeep7 — CHANGES_REQUESTED** [review]

File: .github/workflows/push-to-npm-gpr.yml (hunk 2 above)
Comment (georgedias) [reply .github/workflows/push-to-npm-gpr.yml:35]: Yes you needed it see this url that also covers the scope: https://docs.github.com/en/actions/publishing-packages/publishing-nodejs-packages

File: .github/workflows/push-to-npm-gpr.yml (hunk 1 above)
Comment (georgedias) [reply .github/workflows/push-to-npm-gpr.yml:32]: The file is created in the runner and includes the scope prefix, see these urls 

- https://docs.github.com/en/actions/publishing-packages/publishing-nodejs-packages
- https://docs.npmjs.com/cli/v9/configuring-npm/npmrc

File: .github/workflows/push-to-npm-gpr.yml (hunk 3 above)
Comment (georgedias) [reply .github/workflows/push-to-npm-gpr.yml:36]: https://docs.github.com/en/actions/publishing-packages/publishing-nodejs-packages

File: .github/workflows/push-to-npm-gpr.yml
[the remark this answers is not in the document]
Comment (georgedias) [reply .github/workflows/push-to-npm-gpr.yml:1]: I don't understand your firs ask, there isn't any `remove testing resources` above the `- name: Install project dependencies` line 18

**Review by georgedias — COMMENTED** [review]
Awaiting your input on your first ask.

File: .github/workflows/push-to-npm-gpr.yml
[the remark this answers is not in the document]
Comment (georgedias) [reply .github/workflows/push-to-npm-gpr.yml:1]: Understand the switching the remove testing with install dependencies, done

File: .github/workflows/push-to-npm-gpr.yml (hunk 3 above)
Comment (Amndeep7) [reply .github/workflows/push-to-npm-gpr.yml:36]: how does our current publishing to npm work without the scope variable?  is the scope variable even required?  does it do more under the hood than just prepend the '@mitre' to the package name?

**Review by Amndeep7 — CHANGES_REQUESTED** [review]

**Review by georgedias — COMMENTED** [review]
switched lines - done

File: .github/workflows/push-to-npm-gpr.yml (hunk 2 above)
Comment (georgedias) [reply .github/workflows/push-to-npm-gpr.yml:35]: I will add it, but I don't believe it is necessary as this is set up on the runner and does not haver any interaction with our code or dependencies, so letting the default node-version being used is fair.

File: .github/workflows/push-to-npm-gpr.yml (hunk 3 above)
Comment (georgedias) [reply .github/workflows/push-to-npm-gpr.yml:36]: My take is that the publish to `npm` does not need the scope, if it is working don't mess with it...lol

**Review by georgedias — COMMENTED** [review]
See my comments.

File: .github/workflows/push-to-npm-gpr.yml (hunk 3 above)
Comment (Amndeep7) [reply .github/workflows/push-to-npm-gpr.yml:36]: https://docs.npmjs.com/cli/v9/using-npm/scope#associating-a-scope-with-a-registry

After reading a whole bunch and checking out code, it seems like the scope stuff is multipart.  The first part is that a repo is automatically "scoped" by having "@whatever/" as part of the package name in the package.json, which we do with the "@mitre/".  Second, for npmjs after you publish a package, you can add it to your 'team' there so that it is officially part of your organization.  Third, for github that's not as relevant since these packages are associated with the repos and accounts they're created from.  Fourth, you can locally associate scopes to registries such that when you publish a package it is automatically sent to that particular registry.  However, since we re-initialize that npmrc file every time and point it to a specific registry via the action, I do not think we need to do that local association.

In conclusion, we can safely remove the 'scope' input and it should work.  If it turns out to not do what we want on our next release, we'll know how to fix it, and we could manually push up the release.

File: .github/workflows/push-to-npm-gpr.yml (hunk 2 above)
Comment (Amndeep7) [reply .github/workflows/push-to-npm-gpr.yml:35]: To be clear, I do not think it uses the same runner as the first invocation of the setup-node action.  From the readme: "If not supplied, the node version from PATH will be used. However, it is recommended to always specify Node.js version and don't rely on the system one."  Consequently, I think we should explicitly set the node version here as well.

**Review by Amndeep7 — CHANGES_REQUESTED** [review]

**Review by georgedias — COMMENTED** [review]
made request changes

**Review by Amndeep7 — APPROVED** [review]
